File: programmers/연습문제/해시/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    # participant, completion 모두 정렬
    participant.sort()
    completion.sort()
    answer = participant[-1] # completion과 동일하면 마지막 값이 답이 되기에 미리 저장
    for i in range(len(completion)):
        if completion[i] != participant[i]:
            answer = participant[i]
            break
    return answer


# participant = ["mislav", "stanko", "mislav", "ana"]
# completion = ["stanko", "ana", "mislav"]
# print(solution(participant, completion))

File: programmers/연습문제/해시/완주하지 못한 선수.js
function solution(participant, completion) {
    participant.sort()
    completion.sort()
    let answer = participant[participant.length - 1]
    for (let i = 0; i < completion.length; i ++) {
        if (completion[i] !== participant[i]) {
            answer = participant[i]
            break
        }        
    }
    return answer;
}


// let participant = ["leo", "kiki", "eden"]
// let completion = ["eden", "kiki"]
// console.log(solution(participant, completion))